### PR TITLE
Simple Lasso-Selection

### PR DESCRIPTION
**Pull request by simonbethke**
This is the lasso-selection without reorganizing the code of the selection-tools

https://github.com/user-attachments/assets/e0829bad-6ac6-4fbe-834f-445d2b7452b3

### CODE REVIEW

**Comment by simonbethke** [comment]
When marching ants have been completed, https://github.com/playcanvas/supersplat/pull/266 has to get merged into this

File: src/tools/lasso-selection.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                parent.setPointerCapture(dragId);
+
+                // initialize canvas
+                if (canvas.width !== parent.clientWidth || canvas.height !== parent.clientHeight) {
+                    canvas.width = parent.clientWidth;
+                    canvas.height = parent.clientHeight;
+                }
+
+                // clear canvas
+                context.clearRect(0, 0, canvas.width, canvas.height);
```
Comment (slimbuck) [inline src/tools/lasso-selection.ts:77]: No need to do this here. It gets done in `commitSelection` again.

File: src/tools/lasso-selection.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                    context.moveTo(p.x, p.y);
+                }
+                else {
+                    context.lineTo(p.x, p.y);
+                }
+            });
+            context.closePath();
+            context.fill();
+
+            events.fire(
```
Comment (slimbuck) [inline src/tools/lasso-selection.ts:143]: Two of these are being fired. Please remove either this one or the other.

**Review by slimbuck — APPROVED** [review]
Awesome, thanks so much for this!

**Comment by simonbethke** [comment]
Hmm, I just had fixed your comments and started fighting git. Later I realized that I was fighting the changes that you requested me to do and then did yourself.

**Comment by slimbuck** [comment]
Oh sorry, yeah realised it'd be quicker.

**Pull request merged**